samples/client/petstore/python_disallowAdditionalPropertiesIfNotPresent/petstore_api/model_utils.py

```diff
@@ -186,6 +187,30 @@ def __getattr__(self, attr):
         """get the value of an attribute using dot notation: `instance.attr`"""
         return self.__getitem__(attr)
 
+    def __copy__(self):
+        cls = self.__class__
+        if self.get("_spec_property_naming", False):
+            return cls._new_from_openapi_data(**self.__dict__)
+        else:
+            return new_cls.__new__(cls, **self.__dict__)
+
+    def __deepcopy__(self, memo):
+        cls = self.__class__
+        kwargs = {}
+        if self.discriminator and self.discriminator.keys():
+            key = list(cls.discriminator.keys())[0]
+            kwargs[key] = self.get(key)
+
+        if self.get("_spec_property_naming", False):
+            new_inst = cls._new_from_openapi_data(**kwargs)
+        else:
+            new_inst = cls.__new__(cls, **kwargs)
+
+        for k, v in self.__dict__.items():
+            setattr(new_inst, k, deepcopy(v, memo))
```

Comment:
How about grabbing all of these properties before passing them into 
_new_from_openapi_data and _new_from_openapi_data?
```
kwargs = {}
for k, v in self.__dict__.items():
    kwargs[key] = deepcopy(v, memo)

if self.get("_spec_property_naming", False):
    new_inst = cls._new_from_openapi_data(**kwargs)
else:
    new_inst = cls.__new__(cls, **kwargs)
```

Reply:
The reason setattr is used rather than kwargs, is that the dictionary set of values (such as the properties), are stored in __dict__["_data_store"]. In order to pass as kwargs, I would need to interpret the data (handle the _data_store as a special case), and pass that in as the kwargs. Rather, using setinst sets the key/value as is without any interpretation, a more resilent solution IMO.